Layout Animations - docs (#2062)
software-mansion/react-native-reanimated: docs/docs/api/LayoutAnimations/AnimatedLayout.md

---

> id: layout_animations_animated_layout

Reviewer: I think we don't need to prefix ids with namespace since we have to use paths anyway.

We also use camel case in other docs' ids so this id should be "animatedLayout" and filename should start from lowercase.

title: <AnimatedLayout>
sidebar_label: <AnimatedLayout>
---

> The component is responsible for observing changes in its subtree. It manages all its decendents' animations. Even if it has been removed. There is no need to nest one `AnimatedLayout` in another one. You can treat it as a `View` component.

Reviewer: I'd clarify what it means to be "removed" and merge it with the previous sentence. Why there's no need to nest one AnimatedLayout in another one? This should be explained.

## Example
```js
    import { AnimatedLayout } from 'react-native-reanimated';
    
    function CustomFunctionComponent() {

        return (
            <AnimatedLayout>
                // watched subtree
            </AnimatedLayout>
        );
    }

> ```

Reviewer: We should provide copy-pastable examples of actual use, this example doesn't really show much.

---
id: animatedLayout
title: <AnimatedLayout>
sidebar_label: <AnimatedLayout>
---

> The component is responsible for observing changes in its subtree. There is no need to nest one `AnimatedLayout` in another one because the higher one would already watch the subtree of the lower one. You can treat it as a `View` component. `AnimatedLayout` manages all its descendants' animations while it's mounting and unmounting as well. 

Reviewer 2: "The `AnimatedLayout` component ..."

Reviewer 2: I think it is important to mention that the component should be placed somewhere in the tree in order for enter/exit/layout animations to work. You go straight to explaining that no nesting is needed w/o explaining why it is needed at all

---
id: animatedLayout
title: <AnimatedLayout>
sidebar_label: <AnimatedLayout>
---

> The `AnimatedLayout` component is responsible for observing changes in its subtree `AnimatedLayout`. You can treat it as a `View` component because it takes the same set of properties. The component is essential for entering and exiting animations as well as layout transitions and each animated component that wants to make use of any layout animation has to be placed under an `AnimatedLayout` component. There are two important notes about this component that you should keep in mind: 

Reviewer 2: too many `AnimatedLayout` in the first sentence

Reviewer 2: "must be placed" -> "must be placed directly or indirectly"

title: <AnimatedLayout>
sidebar_label: <AnimatedLayout>
---

> The component is responsible for observing changes in its subtree. It manages all its decendents' animations. Even if it has been removed. There is no need to nest one `AnimatedLayout` in another one because the higher one would already watch the subtree of the lower one. You can treat it as a `View` component.

Reviewer: ```suggestion
The component is responsible for observing changes in its subtree. There is no need to nest one `AnimatedLayout` in another one because the higher one would already watch the subtree of the lower one. You can treat it as a `View` component. `AnimatedLayout` manages all its descendants' animations while it's mounting and unmounting as well. 
```

I think directly mentioning unmounting/mounting is clearer than the previous one since it wasn't clear if it still manages them after removing from subtree.

Reviewer: We may also have to clarify that in order to use AnimatedLayout people must use `Animated.View`.

    }
```

```js
    import { AnimatedLayout } from 'react-native-reanimated';
    
    ​
    const AnimatedText = Animated.createAnimatedComponent(Text);
    ​
    function SWMLogo() {
    ​
        return (
            <AnimatedLayout>
                <Animated.View 

>                     entering={SlideInRight.delay(300)} 

Reviewer: It's not imported anywhere, from where `SlideInRight` came from?
